app/models/user.rb

```diff
@@ -41,6 +41,20 @@ def salary
     salaries.last.try(:amount) || 0
   end
 
+  def salary_type
+    salaries.last.try(:type)
+  end
+
+  def count_day_offs_this_month
```

Comment:
we should cover this method with tests. Also, I think it is too complicated and maybe can be simplified by separating responsibilities in different classes. Need to think about this.